tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp

```diff
@@ -733,351 +714,312 @@ getParamParentNameOffset(const ValueDecl *VD, SourceLoc Cursor) {
+          "Failed to remap declaration to latest snapshot.");
+    }
+  }
 
-  DelayedStringRetriever RelDeclsStream(SS);
-  walkRelatedDecls(VD, [&](const ValueDecl *RelatedDecl, bool UseOriginalBase, bool DuplicateName) {
-    RelDeclsStream.startPiece();
-    {
-      RelDeclsStream<<"<RelatedName usr=\"";
-      SwiftLangSupport::printUSR(RelatedDecl, RelDeclsStream);
-      RelDeclsStream<<"\">";
-      if (isa<AbstractFunctionDecl>(RelatedDecl) && DuplicateName) {
-        // Related decls are generally overloads, so print parameter types to
-        // differentiate them.
-        PrintOptions PO;
-        PO.SkipAttributes = true;
-        PO.SkipIntroducerKeywords = true;
-        PO.ArgAndParamPrinting = PrintOptions::ArgAndParamPrintingMode::ArgumentOnly;
-        XMLEscapingPrinter Printer(RelDeclsStream);
-        if (UseOriginalBase && BaseType) {
-          PO.setBaseType(BaseType);
-          PO.PrintAsMember = true;
-        }
-        RelatedDecl->print(Printer, PO);
-      } else {
-        llvm::SmallString<128> Buf;
-        {
-          llvm::raw_svector_ostream OSBuf(Buf);
-          SwiftLangSupport::printDisplayName(RelatedDecl, OSBuf);
+  ide::walkOverriddenDecls(
+      DInfo.VD,
+      [&](llvm::PointerUnion<const ValueDecl *, const clang::NamedDecl *> D) {
+        Buffer.clear();
+        if (auto VD = D.dyn_cast<const ValueDecl *>()) {
+          if (SwiftLangSupport::printUSR(VD, OS))
+            return;
+        } else {
+          if (clang::index::generateUSRForDecl(
+                  D.get<const clang::NamedDecl *>(), Buffer))
+            return;
         }
-        swift::markup::appendWithXMLEscaping(RelDeclsStream, Buf);
+        Strings.push_back(copyString(Allocator, Buffer));
+        Buffer.clear();
+      });
+  Symbol.OverrideUSRs = copyArray(Allocator, llvm::makeArrayRef(Strings));
+  Strings.clear();
+
+  walkRelatedDecls(DInfo.VD, [&](const ValueDecl *RelatedDecl,
+                                 bool UseOriginalBase, bool DuplicateName) {
+    OS << "<RelatedName usr=\"";
+    SwiftLangSupport::printUSR(RelatedDecl, OS);
+    OS << "\">";
+    if (isa<AbstractFunctionDecl>(RelatedDecl) && DuplicateName) {
+      // Related decls are generally overloads, so print parameter types to
+      // differentiate them.
+      PrintOptions PO;
+      PO.SkipAttributes = true;
+      PO.SkipIntroducerKeywords = true;
+      PO.ArgAndParamPrinting =
+          PrintOptions::ArgAndParamPrintingMode::ArgumentOnly;
+      XMLEscapingPrinter Printer(OS);
+      if (UseOriginalBase && DInfo.BaseType) {
+        PO.setBaseType(DInfo.BaseType);
+        PO.PrintAsMember = true;
       }
-      RelDeclsStream<<"</RelatedName>";
+      RelatedDecl->print(Printer, PO);
+    } else {
+      {
+        llvm::raw_svector_ostream DisplayOS(SpareBuffer);
+        SwiftLangSupport::printDisplayName(RelatedDecl, DisplayOS);
+      }
+      swift::markup::appendWithXMLEscaping(OS, SpareBuffer);
```

Comment:
Missing `SpareBuffer.clear()` ?  Or maybe we could just move SpareBuffer into this callback.

Reply:
Thanks, moved in. I originally thought there may be more cases where we needed another buffer, but it turns out that isn't the case.